pkg/cloud/openstack/assets/deployment.yaml

```diff
@@ -0,0 +1,64 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: openstack-cloud-controller-manager
+  namespace: openshift-cloud-controller-manager
+  labels:
+    app: openstack-cloud-controller-manager
+spec:
+  selector:
+    matchLabels:
+      app: openstack-cloud-controller-manager
+  template:
+    metadata:
+      labels:
+        app: openstack-cloud-controller-manager
+    spec:
+      hostNetwork: true
+      serviceAccount: cluster-cloud-controller-manager
+      priorityClassName: system-cluster-critical
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+        - key: CriticalAddonsOnly
+          operator: Exists
+        - key: node-role.kubernetes.io/master
+          operator: Exists
+          effect: "NoSchedule"
+        - key: node.cloudprovider.kubernetes.io/uninitialized
+          value: "true"
+          effect: NoSchedule
+      containers:
+        - name: openstack-cloud-controller-manager
+          image: docker.io/k8scloudprovider/openstack-cloud-controller-manager:latest
```

Comment:
Is there some API for pulling references from current release? I suggest the 'driver' probably needs to do this directly, as depending on the cloud provider it could potentially use arbitrarily many images.

Reply:
Yes, there is usually a combination of [image stream](https://github.com/openshift/machine-api-operator/blob/master/install/image-references), and preliminary [location](https://github.com/openshift/machine-api-operator/blob/master/install/0000_30_machine-api-operator_01_images.configmap.yaml) for storing resolved image hashes, like a configmap for example. All operator is doing is making sure the operands image matches the one from configmap.